app/controllers/cloud_subnet_controller.rb

```diff
@@ -76,38 +81,45 @@ def create
     case params[:button]
     when "cancel"
       javascript_redirect :action    => 'show_list',
-                          :flash_msg => _("Add of new Subnet was cancelled by the user") % {:model => ui_lookup(:table => 'cloud_subnet')}
+                          :flash_msg => _("Add of new Subnet was cancelled by the user") % {
+                            :model => ui_lookup(:table => 'cloud_subnet')}
 
     when "add"
       @subnet = CloudSubnet.new
       options = form_params
       ems = ExtManagementSystem.find(options[:ems_id])
-      if CloudSubnet.class_by_ems(ems).supports_create?
-        begin
-          CloudSubnet.create_subnet(ems, options)
-          # TODO: To replace with targeted refresh when avail. or either use tasks
-          EmsRefresh.queue_refresh(ManageIQ::Providers::NetworkManager)
-          add_flash(_("Creating Cloud Subnet \"%{subnet_name}\"") % {:subnet_name => options[:name]})
-        rescue => ex
-          add_flash(_("Unable to create Cloud Subnet \"%{subnet_name}\": %{details}") % {
-            :subnet_name => options[:name],
-            :details     => ex}, :error)
-        end
-        @breadcrumbs.pop if @breadcrumbs
-        session[:flash_msgs] = @flash_array.dup if @flash_array
-        javascript_redirect :action => "show_list"
+      options.delete(:ems_id)
+      task_id = ems.create_cloud_subnet_queue(session[:userid], options)
+
+      if task_id.kind_of?(Integer)
+        initiate_wait_for_task(:task_id => task_id, :action => "create_finished")
       else
-        @in_a_form = true
-        add_flash(_(CloudSubnet.unsupported_reason(:create)), :error)
-        drop_breadcrumb(
-          :name => _("Add New Subnet") % {:model => ui_lookup(:table => 'cloud_subnet')},
-          :url  => "/cloud_subnet/new"
+        javascript_flash(
+          :text        => _("Cloud Subnet creation: Task start failed: ID [%{id}]") % {:id => task_id.inspect},
```

Comment:
its unusual to use inspect in code, it's really for debugging purposes. The best would be not to spread it.